hardware_interface/include/hardware_interface/system_interface.hpp

```diff
@@ -187,6 +187,22 @@ class SystemInterface : public rclcpp_lifecycle::node_interfaces::LifecycleNodeI
         info_.thread_priority);
       async_handler_->start_thread();
     }
+
+    if (auto locked_executor = params.executor.lock())
```

Comment:
please add unit test getting the node and verifying the name against something hardcoded